internal/remoteconfig/remoteconfig.go

```diff
@@ -198,32 +196,34 @@ func newClient(config ClientConfig) (*Client, error) {
 // Noop if the client has already started.
 func Start(config ClientConfig) error {
 	var err error
-	startOnce.Do(func() {
-		if !internal.BoolEnv("DD_REMOTE_CONFIGURATION_ENABLED", true) {
-			// Don't start polling if the feature is disabled explicitly
-			return
-		}
-		client, err = newClient(config)
-		if err != nil {
-			return
-		}
-		go func() {
-			ticker := time.NewTicker(client.PollInterval)
-			defer ticker.Stop()
-
-			for {
-				select {
-				case <-client.stop:
-					close(client.stop)
-					return
-				case <-ticker.C:
-					client.Lock()
-					client.updateState()
-					client.Unlock()
-				}
+	if !internal.BoolEnv("DD_REMOTE_CONFIGURATION_ENABLED", true) {
+		// Don't start polling if the feature is disabled explicitly
+		return nil
+	}
+	if !started.CompareAndSwap(false, true) {
```

Comment:
I think we should mark it started when we actually successfully start the client. And I would rather use a `mutex` instead of an atomic boolean. It is easier to reason about and control the in-flight starts.